src/daemon/auth/common/crypt.rs

```diff
@@ -38,10 +38,61 @@ const POLY1305_TAG_BYTE_LEN: usize = POLY1305_TAG_BIT_LEN / 8;
 const CLEARTEXT_PREFIX_LEN: usize = CHACHA20_NONCE_BYTE_LEN + POLY1305_TAG_BYTE_LEN;
 const UNUSED_AAD: [u8; 0] = [0; 0];
 
+//------------ CrossPlatformDeserializableAtomicU64 -------------------------------------------------------------------
+
+/// This type provides Serde (de)serialization irrespective of target platform, as the Serde crate provided derive
+/// impls only appear to produce code when compiled on x86_64. See: https://github.com/NLnetLabs/krill/issues/751
+#[derive(Debug)]
+struct CrossPlatformDeserializableAtomicU64(AtomicU64);
```

Comment:
to me it would be more intuitive to use a notation like `#[serde(deserialize_with = "ext_serde::de_atomic64", serialize_with = "ext_serde::ser_atomicu64")]` instead of a wrapper type. And then copy those functions over to  `commons/utils/ext_serde.rs`.

Reply:
Let me try that.